src/main/scala/chisel3/tester/internal/TestEnvInterface.scala

```diff
@@ -13,6 +18,24 @@ trait TestEnvInterface {
 
   def topFileName: Option[String]
 
+  /**
+    * Will add a TargetDirAnnotation with defaultDir with "test_run_dir" path prefix to the annotations
+    * if there is not a TargetDirAnnotation already present
+    *
+    * @param defaultDir     a default directory
+    * @param annotationSeq  annotations to add it to, unless one is already there
+    * @return
+    */
+  def addDefaultTargetDir(defaultDir: String, annotationSeq: AnnotationSeq): AnnotationSeq = {
+    if(annotationSeq.exists { case _: TargetDirAnnotation => true ; case _ => false }) {
+      annotationSeq
+    }
+    else {
+      val target = TargetDirAnnotation("test_run_dir" + File.separator + defaultDir)
```

Comment:
consider just using the target dir, instead of prepending test_run_dir, especially since you're putting this in a base class where it may be used in other contexts.

Reply:
Not sure what you are suggesting here. Are you saying working directories should end up somewhere other than where they are now, or somehow generate the current ones in a different manner?

Reply:
No, the end result should be the same, it's just a matter of where the test_run_dir gets tacked on. I'd argue that should be at whatever invokes this function, since the test environment might want to define something completely different, but still reuse this annotation creator.